src/mds/CInode.cc

```diff
@@ -4842,7 +4855,11 @@ void CInode::validate_disk_state(CInode::validated_data *results,
           dout(20) << "divergent backtraces are acceptable when dn "
                       "is being purged or has been renamed or moved to a "
                       "different directory " << *in << dendl;
-        }
+        } else if (in->is_any_ancestor_inode_a_replica()) {
+          results->backtrace.passed = true;
+          dout(20) << "divergent backtraces are acceptable when some "
+	              "ancestor inodes are replicas " << *in << dendl;
+	}
```

Comment:
I was wondering if following any path below a replica dir inode is sensible at all. Shouldn't scrub follow the auth inode trail. But that would lead to a distributed algorithm and a complex one I guess.

Reply:
Good question. So, one of the non-auth parent inodes could have a auth cdir (pinned directory) and therefore that subtree needs to be scrubbed. So, if scrub jumps off to the auth-mds of the inode (note that its cdir would be auth), then the entire subtree doesn't get scrubbed.